src/modules-load/modules-load.c

```diff
@@ -52,12 +53,30 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int apply_file(struct kmod_ctx *ctx, const char *path, bool ignore_enoent) {
+static void *worker_thread(void *arg) {
+        int sock_fd = *(int *) arg;
+        char buffer[32];
+        ssize_t bytes_received;
+        _cleanup_(sym_kmod_unrefp) struct kmod_ctx *ctx = NULL;
+        int r = module_setup_context(&ctx);
```

Comment:
in our codebase `int r;` is always declared close to the top of each function, on its own, as it is *the* variable we store error codes in.